presto-main/src/main/java/com/facebook/presto/server/TaskResource.java

```diff
@@ -326,6 +371,12 @@ public void getResults(
         asyncResponse.register((CompletionCallback) throwable -> resultsRequestTime.add(Duration.nanosSince(start)));
     }
 
+    /**
+     * shuffle related
+     *
+     * When the downstream task receives the data, it will (optionally) sends out acknowledgement to the upstream task.
```

Comment:
perhaps, "Acknowledge that the chunk of data for the specified task and partition has been received by deleting that data. This frees up space in the output buffer and allows the task to continue execution and produce more results in case it was blocked due to full buffers."